### PR TITLE
samples: usb: mass Raspberry Pi Pico Support

### PR DESCRIPTION
**Pull request by hlord2000**
Added a flash partition for the mass storage example.

Signed-off-by: Kelly Lord <kellyhlord@gmail.com>

### CODE REVIEW

**Review by tmon-nordic — DISMISSED** [review]

File: samples/subsys/usb/mass/boards/rpi_pico.overlay [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+&flash0 {
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		code_partition: partition@100 {
+			label = "code";
+			reg = <0x100 (DT_SIZE_M(1) - 100)>;
```
Comment (jfischer-no) [inline samples/subsys/usb/mass/boards/rpi_pico.overlay:17]: This looks suspicious, https://github.com/zephyrproject-rtos/zephyr/blob/e364f29bffdfb122fead4b7dad082428f2f6969c/boards/arm/rpi_pico/rpi_pico-common.dtsi#L86-L87

0x100? 256? 100?

**Review by jfischer-no — CHANGES_REQUESTED** [review]

File: samples/subsys/usb/mass/boards/rpi_pico.overlay (hunk 1 above)
Comment (hlord2000) [reply samples/subsys/usb/mass/boards/rpi_pico.overlay:17]: Added reference to RP2040 datasheet for this. Also changed to DT_SIZE_M(2) - 0x100, this was an error.

**Review by tmon-nordic — APPROVED** [review]

**Review by jfischer-no — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

File: samples/subsys/usb/mass/boards/rpi_pico.overlay [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+&flash0 {
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		/*
+		 * Usable flash. Starts at 0x100, after the bootloader. The partition
+		 * size is 1MB minus the 0x100 bytes taken by the bootloader.
```
Comment (fabiobaltieri) [inline samples/subsys/usb/mass/boards/rpi_pico.overlay:17]: Is the "bootloader" wording correct here? Maybe you wanted to say "taken by the firmware" instead.

Comment (raveious) [reply samples/subsys/usb/mass/boards/rpi_pico.overlay:17]: Looks like the comment is copied and updated from the original `rpi_pico` device tree. I think the comment is still correct as the second stage bootloader does go from 0x0 to 0x100.

However, doesn't this file replace the `flash0` node completely? The current Pico device tree has a `second_stage_bootloader` node that represents that boot loader partition, which I think still needs to remain.

**Pull request merged**